qt/widgets/instrumentview/src/BankRenderingHelpers.cpp

```diff
@@ -0,0 +1,153 @@
+#include "MantidQtWidgets/InstrumentView/BankRenderingHelpers.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Objects/IObject.h"
+#include "MantidGeometry/Objects/ShapeFactory.h"
+#include "MantidGeometry/Rendering/GeometryHandler.h"
+#include "MantidGeometry/Rendering/OpenGL_Headers.h"
+#include "MantidGeometry/Rendering/ShapeInfo.h"
+#include "MantidKernel/Quat.h"
+
+namespace {
+Mantid::Kernel::Logger g_log("BankRenderingHelpers");
+
+// Round a number up to the nearest power  of 2
+size_t roundToNearestPowerOfTwo(size_t val) {
+  size_t rounded = 2;
+  while (val > rounded)
+    rounded *= 2;
+  return rounded;
+}
+
+void addVertex(Mantid::Kernel::V3D &pos, double xstep, double ystep) {
```

Comment:
Seems odd that `pos` is being changed at all. Shouldn't this be a `const` you can create a temporary to include the offsets inside the function?